applications/nvidia_nim/nvidia_nim_imaging/app.py

```diff
@@ -0,0 +1,290 @@
+        )
+
+        volume_allocator = UnboundedAllocator(self, name="allocator")
+        logger.info(f"NIFTI file: {self._nifti_file_path}")
+        nifti_volume_loader = VolumeLoaderOp(
+            self,
+            name="nifti_volume_loader",
+            file_name=self._nifti_file_path,
+            allocator=volume_allocator,
+        )
+
+        mask_volume_loader = VolumeLoaderOp(
+            self,
+            CountCondition(self, count=1),
+            name="mask_volume_loader",
+            allocator=volume_allocator,
+        )
+
+        volume_renderer = VolumeRendererOp(
+            self,
+            name="volume_renderer",
+            config_file=self._rendering_config,
+            allocator=volume_allocator,
+            alloc_width=512,
+            alloc_height=512,
+        )
+
+        # Python is not supporting gxf::VideoBuffer, need to convert the video buffer received
+        # from the volume renderer to a tensor.
+        volume_renderer_format_converter = FormatConverterOp(
+            self,
+            name="volume_renderer_format_converter",
+            pool=volume_allocator,
+            in_dtype="rgba8888",
+            out_dtype="rgba8888",
+        )
+        visualizer = HolovizOp(self, name="viz", **self.kwargs("holoviz"))
+
+        self.add_flow(nim_op, mask_volume_loader, {("volume_file", "file_name")})
+        self.add_flow(
+            nifti_volume_loader,
+            volume_renderer,
+            {
+                ("volume", "density_volume"),
+                ("spacing", "density_spacing"),
+                ("permute_axis", "density_permute_axis"),
+                ("flip_axes", "density_flip_axes"),
+            },
+        )
+        self.add_flow(
+            mask_volume_loader,
+            volume_renderer,
+            {
+                ("volume", "mask_volume"),
+                ("spacing", "mask_spacing"),
+                ("permute_axis", "mask_permute_axis"),
+                ("flip_axes", "mask_flip_axes"),
+            },
+        )
+
+        self.add_flow(
+            volume_renderer,
+            volume_renderer_format_converter,
+            {("color_buffer_out", "source_video")},
+        )
+
+        self.add_flow(volume_renderer_format_converter, visualizer, {("tensor", "receivers")})
+
+    def set_nifti_file_path(self, file):
+        self._nifti_file_path = file
+
+    def set_rendering_config(self, file):
+        self._rendering_config = file
+
+
+def _download_dataset(api_key):
+    logger.info("Downloading sample image data from NVIDIA NGC...")
+    headers = {
+        "Authorization": f"Bearer {api_key}",
+    }
+    response = requests.get(payload["image"], headers=headers)
+    response.raise_for_status()
+    nifti_filename = os.path.abspath("sample.nii")
+    buffer = io.BytesIO(response.content)
+    with gzip.GzipFile(fileobj=buffer) as gz_file:
+        with open(nifti_filename, "wb") as f:
+            f.write(gz_file.read())
```

Comment:
I think NIFTI supports compressed files, we could avoid this step.

Reply:
Awesome! Thank you!